src/sections/layer_and_mask_information_section/layer.rs

```diff
@@ -1,17 +1,64 @@
+use std::collections::HashMap;
+
+use failure::{Error, Fail};
+
 use crate::psd_channel::IntoRgba;
 use crate::psd_channel::PsdChannelCompression;
 use crate::psd_channel::PsdChannelError;
 use crate::psd_channel::PsdChannelKind;
 use crate::sections::image_data_section::ChannelBytes;
-use failure::{Error, Fail};
-use std::collections::HashMap;
+
+/// PsdLayerGroup represents a group of layers
```

Comment:
I'm curious about your thoughts on this data structure.

I feel like we should treat the layers as a just bottom to top list of layers .. then have groups just be pointers to some first and last layer. Which is effectively what it is in the spec - just a pointer to the top of one layer and the bottom of another.

Feels more lightweight to be vs. needing to have layers and sub-layers and sub-sub-layers etc.